lib/BlockedBy.js

```diff
@@ -103,6 +103,42 @@ async function blockedBySelf({ waitingKey, buildId, collapse }) {
 
     return false;
 }
+
+/**
+ *
+ * @param {String} buildConfig
+ * @param {Array} blockingBuildIds
+ * @param {String} buildId
+ * @param {Object} redisConn
+ */
+async function checkMultipleBuildsInSameEvent(buildConfig, blockingBuildIds, buildId, redisConn) {
+    if (!buildConfig) return false;
+
+    const currentBuild = JSON.parse(buildConfig);
+
+    if (!currentBuild || !currentBuild.eventId || !currentBuild.jobId) {
+        return false;
+    }
+
+    const isSameBuild = await Promise.race(blockingBuildIds.map(async (id) => {
+        const blockedBuild = JSON.parse(await redisConn.hget(
+            `${queuePrefix}buildConfigs`, id
+        ));
+        const hasEventId = blockedBuild && blockedBuild.eventId && blockedBuild.jobId;
+        const isSameJob = hasEventId && (currentBuild.eventId === blockedBuild.eventId)
+            && (currentBuild.jobId === blockedBuild.jobId);
+
+        if (isSameJob) {
```

Comment:
there is one legit use case for same job multiple builds https://github.com/screwdriver-cd/screwdriver/issues/874#issuecomment-468735515
This change will force `~` builds into option 2. But I think it's okay for now, UI anyways can't handle option 1, and when we feature complete https://github.com/screwdriver-cd/screwdriver/issues/874 we can revisit this check to include a common `parentBuildID` check as well.

cc @tkyi